src/App.tsx

```diff
@@ -1,61 +1,105 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+
+const getUserById = (userId: number) => {
+  return usersFromServer.find(user => user.id === userId) || null;
+};
+
+const listTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(listTodos);
+  const [title, setTitle] = useState('');
+  const [isTitleError, setIsTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [isUserError, setIsUserError] = useState(false);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleError(false);
+  };
+
+  const handleChangeUserId = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setIsUserError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const onAdd = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const newUser = getUserById(userId) || null;
+
+    if (!title || !userId) {
```

Comment:
The condition here checks if either 'title' or 'userId' is falsy. However, this will set both 'isTitleError' and 'isUserError' to true even if only one of them is missing. Consider separating these checks to provide more accurate error messages.